blog/2023-07-13-healthsearch-blogpost/index.mdx

```diff
@@ -0,0 +1,379 @@
+---
+title: Discover Healthsearch Unlocking Health with Semantic Search ✨
+slug: healthsearch-demo
+authors: edward
+date: 2023-07-13
+image: ./img/hero.png
+tags: ['concepts','engineering','how-to']
+description: "Learn about our latest open-source demo and how we used Semantic and Generative Search to improve access to health"
+
+---
+
+![Healthsearch Demo](./img/hero.png)
+
+<!-- truncate -->
+
+## 🛠️ Bridging semantic and generative search!
+
+Isn’t it annoying when you just want to find a remedy for your headache and instead find out that you have a horrible disease? With Healthsearch you will find exactly and only that - and no unsolicited health diagnosis.
+
+Try out the [live demo here](https://healthsearch-frontend.onrender.com/)✨
+
+<figure>
+<img
+    src={require('./img/healthsearch.gif').default}
+    alt="alt"
+    style={{ width: "100%" }}
+/>
+
+:::danger Please note
+Healthsearch is a technical demonstration and acts as a proof of concept.
+The results shown in Healthsearch should not be taken as health advice or is it the intention of this demo. The search results are based on the semantic similarity between the query and user-written reviews.
+:::
+</figure>
+
+Imagine just being able to type, 'I need better sleep,' and instantly get a list of related supplements tailored to enhance your sleep quality. Or you're dealing with persistent joint pain, and you're looking for a natural supplement to alleviate discomfort.
+
+We are super excited to release our latest open-source demo, Healthsearch. This demo decodes user-written reviews of supplements and performs semantic- and generative search on them, retrieving the most related products for specific health effects based on the reviews, and leveraging Large Language Models to generate product and review summaries. The demo can understand natural language queries and derive all search filters directly from the context of your query.
+
+In this blog post, I’ll give you a quick tour of the Healthsearch project and its exciting features. We will also go through the implementations and the challenges I faced while working on it.
```

Comment:
-> In this blog post, I’ll give you a quick tour of the Healthsearch project and its amazing features. We will also go through the implementations and the challenges I faced while working on it.

you used excited already one sentence before this one

Reply:
ahh lets keep this in mind for HN so talking about the faced technical problem of OpenAI API not responding and how caching solves this to some extend